### PR TITLE
enhancement shape analysis pass

### PR DESCRIPTION
**Pull request by Yancey1989**
related #473

### CODE REVIEW

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp [hunk 1]
```diff
@@ -1796,6 +1818,20 @@ class ShapePropagator : public PropertyPropBase {
...
         return true;
       }
+    } else if (
+        node->matches(
+            "aten::embedding_dense_backward(Tensor grad_output, Tensor indices, int num_weights, int padding_idx, bool scale_grad_by_freq) -> Tensor")) {
+      auto grad_output_type = input_type(0);
+      auto maybe_num_weights = node->get<int>(attr::num_weights);
+      if (grad_output_type && maybe_num_weights &&
+          grad_output_type->sizes().size()) {
+        auto idx = *grad_output_type->sizes().size() - 1;
```
Comment (fortianyou) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp:1828]: unused?

File: pytorch_blade/src/compiler/jit/torch/shape_analysis_test.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include <torch/script.h>
+#include "compiler/jit/torch/shape_analysis.h"
+
+namespace torch {
+namespace blade {
+
+TEST(PropagateInputShapesTest, SimpleUnary) {
+#if PYTORCH_MAJOR_VERSOIN == 1 && PYTORCH_MINOR_VERSION >= 8
+  const std::string graph_str = R"IR(
+graph(%p1 : Float(*, *, *, device=cuda:0)):
```
Comment (fortianyou) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis_test.cpp:24]: Will it run successfully on cpu version pytorch?

Comment (Yancey1989) [reply pytorch_blade/src/compiler/jit/torch/shape_analysis_test.cpp:24]: it works well, shape analysis pass does care about the device.

**Review by fortianyou — APPROVED** [review]
LGTM

**Pull request merged**